contracts/exchange/TokenPriceProvider.sol

```diff
@@ -36,17 +20,39 @@ contract TokenPriceProvider {
      * @param _token the ERC20 token contract
      * @return the ether value (in wei) of _amount tokens with contract _token
      */
-    function getEtherValue(uint256 _amount, address _token) public view returns (uint256) {
+    function getEtherValue(uint256 _amount, address _token) external view returns (uint256) {
         uint256 decimals = ERC20(_token).decimals();
         uint256 price = cachedPrices[_token];
         return price.mul(_amount).div(10**decimals);
     }
 
     //
-    // Internal
+    // The following is added to be backward-compatible with Argent's old backend
     //
 
-    function kyberNetwork() internal view returns (KyberNetwork) {
-        return KyberNetwork(KYBER_NETWORK_ADDRESS);
+    // Mock token address for ETH
+    address constant internal ETH_TOKEN_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    // Address of the KyberNetwork contract
+    KyberNetwork public kyberNetwork;
+
+    constructor(KyberNetwork _kyberNetwork) public {
+        kyberNetwork = _kyberNetwork;
+    }
+
+    function setKyberNetwork(KyberNetwork _kyberNetwork) external onlyManager {
+        kyberNetwork = _kyberNetwork;
+    }
```

Comment:
Can we have this at the beginning of the contract? I know it's for backward compatibility with the backend but having the constructor and the storage in the middle makes the contract unclean.

Reply:
Done